### DISCUSSION TITLE
How to do not receive own messages?

### DISCUSSION THREAD

**Discussion by ZedZipDev** [rabbitmq/rabbitmq-dotnet-client#1119] [Ideas]
How to implement this approach: 
I have 3 clients applications. Each of them is producer and consumer. 
They generate messages to one queue (it is now, can be changed if need). 
A consumer should receive messages from all producers excluding its own. 
How to do it? How to exclude own messages? Do not receive from the queue or receive and filter on the client side?

**Comment by lukebakken** [comment]
RabbitMQ will round-robin messages to consumers. It pays no attention to where they came from.

One option is, when a consumer receives its own message, it can `nack` it. RabbitMQ _should_ deliver it to a different consumer.
